### DISCUSSION TITLE
Is there a way to set/update the auth url on the fly?

### DISCUSSION THREAD

**Discussion by notmatthewa** [PaloAltoNetworks/docusaurus-openapi-docs#795] [Q&A]
I'm currently working on integrating custom authentication functionality to accommodate multiple client-specific API URLs within our setup. As part of this, I need to dynamically update the "Auth URL" displayed in the sidebar on each API page to reflect the correct URL for different clients.

After exploring the theme's code, I noticed a middleware action named `server/setServer`, which seems to be relevant to what I'm trying to achieve. However, as I'm not deeply experienced with React, I'm uncertain about the best way to utilize this action for my purpose.

Is my approach of using server/setServer to dynamically change the Auth URL on the fly the correct path to follow? I would greatly appreciate any guidance or insight on how to properly implement this functionality or if there's an intended method for updating the Auth URL that I might have overlooked.

Thank you for your time and help!

**Comment by alienzhangyw** [comment]
You can swizzle and modify the code here:
https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/blob/d2ff97b0ee8ec4da06348a1abcabe5ea8fe55eba/packages/docusaurus-theme-openapi-docs/src/theme/ApiItem/index.tsx#L95

You can change the servers or modify the urls to whaterver.

For example, I replaced the `VAR::DOMAIN_NAME` from api yaml to actual domain set in `siteConfig.customFields.brandConfig`.
```typescript
const { domainName } = siteConfig.customFields.brandConfig as any;
const servers = api?.servers?.map((server) => ({ ...server, url: server.url?.replace('VAR::DOMAIN_NAME', domainName) })) ?? [];
```

**Reply by notmatthewa** [reply]
I will do this. I actually stumbled on swizzling, but wasn't sure if it was the move. Thank you for this!

**Reply by notmatthewa** [reply]
I might be missing something, but if i attempt to swizzle, the moment i re-build the docs, it gives me an error about missing a provider. I think i'm missing something. I used the `yarn swizzle` command and followed [this tutorial](https://docusaurus.io/docs/swizzling).

**Comment by alienzhangyw** [comment]
> I might be missing something, but if i attempt to swizzle, the moment i re-build the docs, it gives me an error about missing a provider. I think i'm missing something. I used the `yarn swizzle` command and followed [this tutorial](https://docusaurus.io/docs/swizzling).

You have to swizzle @theme/ApiExplorer and @theme/ApiItem
